app/services/subscriptions/validate_service.rb

```diff
@@ -35,12 +36,46 @@ def valid_plan?
     end
 
     def valid_subscription_at?
-      return true if args[:subscription_at].respond_to?(:strftime)
-      return true if args[:subscription_at].is_a?(String) && DateTime._strptime(args[:subscription_at]).present?
+      return true if is_valid_format?(args[:subscription_at])
 
       add_error(field: :subscription_at, error_code: 'invalid_date')
 
       false
     end
+
+    def valid_ending_at?
+      return true if args[:ending_at].blank?
+
+      if is_valid_format?(args[:ending_at]) &&
+         is_valid_format?(args[:subscription_at]) &&
+         ending_at.to_date > Time.current.to_date &&
+         ending_at.to_date > subscription_at.to_date
+        return true
+      end
+
+      add_error(field: :ending_at, error_code: 'invalid_date')
+
+      false
+    end
+
+    def is_valid_format?(datetime)
+      datetime.respond_to?(:strftime) || datetime.is_a?(String) && DateTime._strptime(datetime).present?
```

Comment:
Maybe we could move this logic into a small utility class / module. It ssems we are using the same kind of logic in few places and it could probably be shared. What do you think?